test/model.test.js

```diff
@@ -6756,6 +6755,31 @@ describe('Model', function() {
         );
 
       });
+      xit('creates indexes only when they do not exist on the mongodb server (gh-12250)', async() => {
```

Comment:
This is marked as pending because this is not fixed yet, the PR was initially meant to fix the issue, but it's taking longer than anticipated. So for now let's stick to the refactoring, and I'll fix it later